### PR TITLE
added Array Extension Methods to the API page

### PR DESCRIPTION
**Pull request by crswll**
Let me know what you think!

### CODE REVIEW

File: src/api/index.md [hunk 1]
```diff
@@ -2161,3 +2161,39 @@ type: api
...
+  ``` js
+    vm.animals.$set(0, { name: 'Aardvark' })
+  ```
+
+- **See also:** [Mutation Methods](/guide/list.html#Mutation-Methods)
+
+### array.$remove(indexOrReference)
+
+- **Arguments**
+  - `{Number | Reference} indexOrReference`
```
Comment (kazupon) [inline src/api/index.md]: I think that the below jsdoc syntax correct.
`{Number | Reference}` -> `{*}`

**Comment by yyx990803** [comment]
$remove only supports removal by reference now.

For index removal just use splice.

**Comment by crswll** [comment]
Sweet, adjusted.

**Comment by yyx990803** [comment]
Thanks!

**Pull request merged**